examples/pytorch/object-detection/run_object_detection.py

```diff
@@ -0,0 +1,496 @@
+                "Use --overwrite_output_dir to overcome."
+            )
+        elif checkpoint is not None and training_args.resume_from_checkpoint is None:
+            logger.info(
+                f"Checkpoint detected, resuming training at {checkpoint}. To avoid this behavior, change "
+                "the `--output_dir` or add `--overwrite_output_dir` to train from scratch."
+            )
+
+    # ------------------------------------------------------------------------------------------------
+    # Load dataset, prepare splits
+    # ------------------------------------------------------------------------------------------------
+
+    dataset = load_dataset(data_args.dataset_name, cache_dir=model_args.cache_dir)
+
+    # If we don't have a validation split, split off a percentage of train as validation
+    data_args.train_val_split = None if "validation" in dataset.keys() else data_args.train_val_split
+    if isinstance(data_args.train_val_split, float) and data_args.train_val_split > 0.0:
+        split = dataset["train"].train_test_split(data_args.train_val_split)
+        dataset["train"] = split["train"]
+        dataset["validation"] = split["test"]
+
+    # Get dataset categories and prepare mappings for label_name <-> label_id
+    categories = dataset["train"].features["objects"].feature["category"].names
+    id2label = dict(enumerate(categories))
+    label2id = {v: k for k, v in id2label.items()}
+
+    # ------------------------------------------------------------------------------------------------
+    # Load pretrained config, model and image processor
+    # ------------------------------------------------------------------------------------------------
+
+    common_pretrained_args = {
+        "cache_dir": model_args.cache_dir,
+        "revision": model_args.model_revision,
+        "token": model_args.token,
+        "trust_remote_code": model_args.trust_remote_code,
+    }
+    config = AutoConfig.from_pretrained(
+        model_args.config_name or model_args.model_name_or_path,
+        label2id=label2id,
+        id2label=id2label,
+        **common_pretrained_args,
+    )
+    model = AutoModelForObjectDetection.from_pretrained(
+        model_args.model_name_or_path,
+        config=config,
+        ignore_mismatched_sizes=model_args.ignore_mismatched_sizes,
+        **common_pretrained_args,
+    )
+    image_processor = AutoImageProcessor.from_pretrained(
+        model_args.image_processor_name or model_args.model_name_or_path,
+        size={"longest_edge": data_args.longest_edge, "shortest_edge": data_args.shortest_edge},
+        **common_pretrained_args,
+    )
+
+    # ------------------------------------------------------------------------------------------------
+    # Define image augmentations and dataset transforms
+    # ------------------------------------------------------------------------------------------------
+
+    train_augmentation_transform = A.Compose(
```

Comment:
FYI DETR (and subsequent models) typically apply so-called scale augmentation during training: https://github.com/facebookresearch/detr/blob/29901c51d7fe8712168b8d0d64351170bc0f83e0/datasets/coco.py#L115-L144. This means that they use random scales for the resize operation

Reply:
Yes, I think we can also add it. There are different crop variations in the albumenentations.

I also found an interesting thing. Doing manual padding and resizing outside of the image_processor yields around 0.1 better mAP on cppe5 dataset (e.g. 0.7 -> 0.8). For all detection models in transformers. I will check this out and sum up as soon as experiments finish.

Reply:
What I also found, is that if we train with `image_processor` padding + `pixel_mask`, our outputs are batch-dependent, because images are padded to the max height/width in a batch. Evaluation with batch size = 1 also drops mAP to 0.05 compared to evaluation with batch size > 8 (because the model was trained with this batch size).

Reply:
Probably, this effect will not be observed on huge datasets with thousands of images (like COCO), or if we train with a small batch size per GPU, but for fine-tuning might be significant.